src/platform_linux/platform.c

```diff
@@ -137,24 +137,19 @@ platform_thread_join(platform_thread thread)
 }
 
 platform_status
-platform_mutex_init(platform_mutex    *mu,
+platform_mutex_init(platform_mutex    *lock,
                     platform_module_id UNUSED_PARAM(module_id),
                     platform_heap_id   UNUSED_PARAM(heap_id))
 {
-   int ret;
-
-   ret = pthread_mutex_init(mu, NULL);
-
+   int ret     = pthread_mutex_init(&lock->mutex, NULL);
+   lock->owner = INVALID_TID;
    return CONST_STATUS(ret);
 }
 
 platform_status
-platform_mutex_destroy(platform_mutex *mu)
+platform_mutex_destroy(platform_mutex *lock)
 {
-   int ret;
-
-   ret = pthread_mutex_destroy(mu);
-
+   int ret = pthread_mutex_destroy(&lock->mutex);
```

Comment:
For hygiene, better to also do `lock->owner = INVALID_TID;` before returning. No ? Will keep callers safe.